clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -1026,14 +1025,11 @@ invariant_checker_t::check_for_pc_discontinuity(
             }
             if (have_branch_target && branch_target != cur_pc)
                 error_msg = "Branch does not go to the correct target";
-        } else if (cur_instr_decoded != nullptr &&
-                   shard->prev_instr_decoded_ != nullptr &&
+        } else if (cur_instr_decoded != nullptr && prev_instr_info.has_decoded &&
                    instr_is_syscall(cur_instr_decoded->data) &&
```

Comment:
This should be using the new instr info.  Xref the other comment on expecting the cur instr to also use the new type.